test/clear-clipboard.specs.js

```diff
@@ -0,0 +1,20 @@
+const ClearClipboard = require('../src/clear-clipboard')
+const clipboardy = require('clipboardy')
+let assert = require('assert')
+
+describe('clear-clipboard', () => {
+  describe('clearClipboardHistory', () => {
+    beforeEach(() => {
+      clipboardy.writeSync('general kenboi')
+      ClearClipboard.clearClipboardHistory()
+    })
+
+    it('should have cleared the clip board', function(done) {
+      this.timeout(6100)
+      setTimeout(function() {
+        assert.strictEqual(clipboardy.readSync(), '')
+        done()
+      }, 6000)
```

Comment:
It would be nice if we could figure out a way to test this better so we don't have to wait 6 seconds every time -- which is especially painful if you're trying to TDD running changes on a watch (`npm run watch`).

Could we maybe override the `numberOfItemsHistoryStores` to a much lower number when testing? Or mock out `clipboardy` so we don't have to wait 180ms between writes?

Also, is there a way to verify that this has cleared the whole history? We've set a magic number of 25 history items, but this test would pass if it only set a single clipboard value, right?

Reply:
Properly testing this is something I've been struggling with. I'm not sure if there is a way to verify that the entire history is cleared.
Shortening the test run time by overriding the `numberOfItemsHistoryStores` should be doable, and maybe I could assert that `writeToClipboard` was called `numberOfItemsHistoryStores` times? I'm not sure if that is possible here, but I can look into it.

Reply:
Take a look at `state-persister.specs.js` for an example of how we used `sinon` to stub out the `fs` filesystem module. We really don't need to test clipboardy itself, we know it should set the clipboard value. So mocking it out makes sense for our case -- allowing you to check that it did get called the right number of times.